qiskit/providers/ibmq/utils/utils.py

```diff
@@ -56,3 +58,65 @@ def validate_job_tags(job_tags: Optional[List[str]], exception: Type[Exception])
     if job_tags and (not isinstance(job_tags, list) or
                      not all(isinstance(tag, str) for tag in job_tags)):
         raise exception("job_tags needs to be a list or strings.")
+
+
+class RefreshQueue(Queue):
+    """A queue that replaces the oldest item with the new when full.
```

Comment:
Swapping `new` for `newest item` might make more sense in this sentence

Reply:
I didn't want to use `newest item` because that implies the item is already on the queue and not being added to the queue. Reworded it to hopefully make it clearer.